clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,25 +25,40 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
     private final List<String> topics;
 
+    /**
+     * Constructor to support asking for no topic metadata
+     */
+    public MetadataRequest() {
```

Comment:
@hachikuji @ijuma I updated the constructor to take an `allTopics` boolean. I think the changes required to do so may also solve [KAFKA-3358](https://issues.apache.org/jira/browse/KAFKA-3358). It worked nicely with `metadata.needMetadataForAllTopics` in the `NetworkClient`.